roles/setup_barman/tasks/setup_barman.yml

```diff
@@ -0,0 +1,58 @@
+---
+
+- name: Set the os variable
+  set_fact:
+    os: "{{ ansible_distribution | replace('RedHat', 'RHEL') }}{{ ansible_distribution_major_version }}"
+
+- name: Set barman_backup_method
+  set_fact:
+    barman_backup_method: "{{ hostvars[ansible_hostname].barman_backup_method | default('rsync') }}"
+
+- name: Check barman_backup_method value
+  fail:
+    msg: "Barman backup method {{ barman_backup_method }} not supported."
+  when: >
+    barman_backup_method not in ['postgres', 'rsync']
+
+- name: Check support for Operating System
+  fail:
+    msg: "Operating System = {{ os }} not supported."
+  when: os not in supported_os
+
+- name: Reference pg_type variables
+  include_vars: "../../init_dbserver/vars/{{ pg_type }}.yml"
```

Comment:
Do you think it's a good practice for using relative paths for variables?

Reply:
Done. Replaced by `include_vars: "{{ role_path }}/../<antoher_role>/vars/<...>"`